functions/Connect-VcdOrganisation.ps1

```diff
@@ -4,53 +4,54 @@ function Connect-VcdOrganisation {
         [Parameter(Mandatory = $true)][string]$Organisation,
         [Parameter(Mandatory = $true)][System.Management.Automation.PSCredential]$Credential,
         [Parameter(Mandatory = $true)][string]$APIurl,
-        [Parameter(Mandatory = $false)][string]$APIVersion = "5.1"
+        [Parameter(Mandatory = $false)][string]$APIVersion = "30.0"
     )
 
     try {
-        #region Generate encoded Authentication string
+        #region Generate authentication string
         $AuthenticationString = $($Credential.UserName) + '@' + $Organisation + ':' + $($Credential.GetNetworkCredential().Password)
         $EncodedAuthenticationString = [System.Text.Encoding]::UTF8.GetBytes($AuthenticationString)
         Remove-Variable AuthenticationString
         $EncodedAuthenticationStringBase64 = [System.Convert]::ToBase64String($EncodedAuthenticationString)
         Write-Verbose "Generated encoded Authentication string"
         #endregion
 
-        #region Define REST header
-        $headers = @{"Accept" = "application/*+xml;version=$APIVersion"}
-        Write-Verbose "Generated header"
-        #endregion
-
-        #region Get Login Url
+        #region Get login URL
         $Uri = $APIurl + '/versions'
-        $versions = Invoke-RestMethod -Uri $Uri -Headers $headers -Method GET -ErrorAction Stop
+        $versions = Invoke-RestMethod -Uri $Uri -Method GET -ErrorAction Stop
 
         Write-Verbose "Looking for API Version $APIVersion"
         foreach ($VersionInfo in $versions.SupportedVersions.VersionInfo) {
+            #$VersionInfo
             if ($VersionInfo.Version -eq $APIVersion) {
                 $loginUrl = $VersionInfo.LoginUrl
                 Write-Verbose "Login Url: $loginUrl"
                 break
             }
         }
-
         if ([string]::IsNullOrEmpty($loginUrl)) {
             throw "No Login URL Available - Exit"
         }
         #endregion
 
-        #region Login and generate Session
-        # Add Authorization to header
+
+        #region Generate session headers
+        $headers = @{"Accept" = "application/*+xml;version=$APIVersion"}
         $headers += @{"Authorization" = "Basic $($EncodedAuthenticationStringBase64)"}
-        $SessionInformation = Invoke-RestMethod -Uri $loginurl -Headers $headers -Method POST -Session vCloudSession -ErrorAction Stop
-        if ($vCloudSession.Headers.Count -eq 0 ) {
+        Write-Verbose "Generated header"
+        [xml]$session

```

Comment:
Do you use this anywhere? If not please remove it. Best to check with PowerShell Script Analyzer